CMakeLists.txt

```diff
@@ -413,22 +394,56 @@ if(${PLAYER_TARGET_PLATFORM} STREQUAL "SDL2")
 	target_link_libraries(${PROJECT_NAME} SDL2::SDL2main)
 elseif(${PLAYER_TARGET_PLATFORM} STREQUAL "libretro")
 	target_sources(${PROJECT_NAME} PRIVATE
-		src/audio_libretro.cpp
-		src/audio_libretro.h
-		src/input_buttons_libretro.cpp
-		src/libretro_ui.cpp
-		src/libretro_ui.h)
+		src/platform/libretro/libretro_audio.cpp
+		src/platform/libretro/libretro_audio.h
+		src/platform/libretro/libretro_input_buttons.cpp
+		src/platform/libretro/libretro_ui.cpp
+		src/platform/libretro/libretro_ui.h)
 	target_compile_definitions(${PROJECT_NAME} PUBLIC USE_LIBRETRO=1)
 	set(PLAYER_BUILD_EXECUTABLE OFF)
 	set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "")
 	set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME "easyrpg_libretro")
 
 	add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/builds/libretro)
 	target_link_libraries(${PROJECT_NAME} retro_common)
+elseif(${PLAYER_TARGET_PLATFORM} STREQUAL "3ds")
+	target_sources(${PROJECT_NAME} PRIVATE
+		src/platform/3ds/3ds_audio.cpp
+		src/platform/3ds/3ds_audio.h
+		src/platform/3ds/3ds_clock.cpp
+		src/platform/3ds/3ds_clock.h
+		src/platform/3ds/3ds_ui.cpp
+		src/platform/3ds/3ds_ui.h)
+	target_compile_definitions(${PROJECT_NAME} PUBLIC _3DS)
+elseif(${PLAYER_TARGET_PLATFORM} STREQUAL "psvita")
+	target_sources(${PROJECT_NAME} PRIVATE
+		src/platform/psvita/psp2_audio.cpp
+		src/platform/psvita/psp2_audio.h
+		src/platform/psvita/psp2_clock.cpp
+		src/platform/psvita/psp2_clock.h
+		src/platform/psvita/psp2_ui.cpp
+		src/platform/psvita/psp2_ui.h)
+	target_compile_definitions(${PROJECT_NAME} PUBLIC PSP2)
+elseif(${PLAYER_TARGET_PLATFORM} STREQUAL "switch")
+	target_sources(${PROJECT_NAME} PRIVATE
+		src/platform/switch/switch_audio.cpp
+		src/platform/switch/switch_audio.h
+		src/platform/switch/switch_clock.cpp
+		src/platform/switch/switch_clock.h
+		src/platform/switch/switch_ui.cpp
+		src/platform/switch/switch_ui.h)
```

Comment:
Doesn't have to happen now, but should be call this `switch/ui.h` etc.. instead of duplicating the platform name twice?

Reply:
>  drop double 3ds/3ds_ and such prefix (currently not possible since dkp Makefiles need distinct file names)

Reply:
Visual Studio project files have a similiar problem. At least in VS2017 the default name for object files was without the directory. No idea if they fixed this by now though (and CMake is likely smart enough and overwrites this default)